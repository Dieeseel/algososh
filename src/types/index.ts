import { ElementStates } from "./element-states"

export type TLetter = {
    letter: string | undefined
    state: ElementStates
}

export type TColumn = {
    number: number
    state: ElementStates
}

export type TQueue<T> = {
    elements: (T | null)[], 
    head: number, 
    tail: number
}


export interface IStack<T> {
    push: (item: T) => void;
    pop: () => void;
    getElements: () => void
    clearStack: () => void
}

export interface IQueue<T> {
    enqueue: (item: T) => void;
    dequeue: () => void;
    clear: () => void
    getElements: () => TQueue<T>
    isEmpty: () => boolean
}

export interface ILinkedList<T> {
    append: (element: T) => void;
    getSize: () => number;
    getArray: () => T[];
}