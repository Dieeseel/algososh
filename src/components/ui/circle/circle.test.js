import renderer from 'react-test-renderer';
import { ElementStates } from '../../../types/element-states';
import { cleanup } from '@testing-library/react';
import { Circle } from './circle';


afterEach(cleanup);


describe('Тест рендера компонента Circle', () => {
  it('Рендер компонента Circle без буквы прошел без ошибок', () => {
    const circle = renderer
      .create(<Circle letter='' />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle с буквами прошел без ошибок', () => {
      const circle = renderer
        .create(<Circle letter='abc' />)
        .toJSON();
        expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle с head прошел без ошибок', () => {
      const circle = renderer
        .create(<Circle head='head' />)
        .toJSON();
        expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle с react-элементом в head; прошел без ошибок', () => {
      const circle = renderer
        .create(<Circle head={<Circle letter='a' isSmall={true} />} />)
        .toJSON();
        expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle с tail прошел без ошибок', () => {
    const circle = renderer
      .create(<Circle tail='tail' />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle с react-элементом в tail; прошел без ошибок', () => {
    const circle = renderer
      .create(<Circle tail={<Circle letter='a' isSmall={true} />} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle с index прошел без ошибок', () => {
    const circle = renderer
      .create(<Circle index={1} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle с пропcом isSmall === true прошел без ошибок', () => {
    const circle = renderer
      .create(<Circle isSmall={true} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle в состоянии default прошел без ошибок', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle в состоянии changing прошел без ошибок', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Рендер компонента Circle в состоянии modified прошел без ошибок', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });
})