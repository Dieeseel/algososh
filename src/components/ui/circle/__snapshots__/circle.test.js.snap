// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест рендера компонента Circle Рендер компонента Circle без буквы прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle в состоянии changing прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle в состоянии default прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle в состоянии modified прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle с head прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle с index прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle с react-элементом в head; прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          a
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle с react-элементом в tail; прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          a
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle с tail прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle с буквами прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle Рендер компонента Circle с пропcом isSmall === true прошел без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
